guidelines/mwm-storage.md

```diff
@@ -1,3 +1,30 @@
-# Storage handling
+# Storage
 
+Input data, metadata & task output artifacts are all stored in a shared storage layer.
+
+### Architecture Diagram
+![store-arch.png](static/store-arch.png)
+
+### Storage Backend
+
+For the first releases (TBD) storage will be any S3 compatible resource. In a cloud-ready environment this can be AWS S3, and in environments restricted to on-premise storage it can be local services such as [Minio](http://min.io).
```

Comment:
We may need to unify the abstraction layer between modules, even with different programming languages.

Reply:
Yes, this will be implemented as an abstract api with "S3" being a specific backend. For now it'll be the only backend, but that could change going forward. I don't think I know enough to fully define this API right now though.

Reply:
It is ok to not have the API defined now. We don't know what we don't know. 

But +1 to add a line saying end goal is providing an API as abstraction layer between modules. Readers need to know and trust we thought about it and is top of mind. Trusting the design drives adoption. :)

Reply:
Added clarification